swarm/api/http/server.go

```diff
@@ -891,6 +914,7 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *http.Request) {
 	}
 
 	w.Header().Set("Content-Type", contentType)
+	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", path.Base(r.URL.Path)))
```

Comment:
same as above

Reply:
please amend this - use `uri` from context instead of request URI. also - provide the filename fallback